src/app/packages/imagery/imageryComponent/imagery.component.spec.ts

```diff
@@ -5,40 +5,56 @@ import {async, ComponentFixture, inject, TestBed} from '@angular/core/testing';
 import { ImageryComponent } from './imagery.component';
 import {ImageryCommunicatorService} from '../api/imageryCommunicatorService';
 import {ImageryComponentSettings} from './imageryComponentSettings';
+import { MapSettings } from './mapSettings';
+import { ImageryProviderService } from '../imageryProviderService/imageryProviderService';
 
 describe('ImageryComponent', () => {
 	let component: ImageryComponent;
 	let fixture: ComponentFixture<ImageryComponent>;
 	let imageryCommunicatorService: ImageryCommunicatorService;
+	let imageryProviderService: ImageryProviderService;
 
-	const imageryData: ImageryComponentSettings = {mapComponentId: 'imagery1', mapTypes: ['openLayers']};
+	const geoPoint: GeoJSON.Point = {
+		type: 'Point',
+		coordinates: [15.7, 37.9]
+	};
+
+	const mapSettings: MapSettings = { mapType: 'openLayerMap', mapModes: []};
+
+	const imageryData: ImageryComponentSettings = {mapComponentId: 'imagery1', mapSettings: [mapSettings], mapCenter: geoPoint};
 
 	beforeEach(async(() => {
-		TestBed.configureTestingModule({declarations: [ ImageryComponent ], providers: [ ImageryCommunicatorService ]}).compileComponents();
+		TestBed.configureTestingModule({declarations: [ ImageryComponent ], providers: [ ImageryCommunicatorService, ImageryProviderService ]}).compileComponents();
 	}));
 
-	beforeEach(inject([ImageryCommunicatorService], (_imageryCommunicatorService) => {
+	beforeEach(inject([ImageryCommunicatorService, ImageryProviderService], (_imageryCommunicatorService, _imageryProviderService) => {
 		imageryCommunicatorService = _imageryCommunicatorService;
+		imageryProviderService = _imageryProviderService;
 		fixture = TestBed.createComponent(ImageryComponent);
 		component = fixture.componentInstance;
 		fixture.detectChanges();
 	}));
 
 	it('should create', () => {expect(component).toBeTruthy(); });
 
-	it('Create OPenLayersMap', () => {
-		component.mapComponentSettings = imageryData;
-		component.ngOnInit();
-		const div = fixture.nativeElement.querySelector('#openLayersMap');
-		expect(div).toBeDefined();
-		const olOverlaycontainer = div.querySelector('.ol-overlaycontainer');
-		expect(olOverlaycontainer).toBeDefined();
-	});
+	// it('Create OpenLayersMap', () => {
+	// 	component.mapComponentSettings = imageryData;
+	// 	imageryProviderService.registerMapProvider()
+	// 	component.ngOnInit();
+	// 	const div = fixture.nativeElement.querySelector('#openLayersMap');
+	// 	expect(div).toBeDefined();
+	// 	const olOverlaycontainer = div.querySelector('.ol-overlaycontainer');
+	// 	expect(olOverlaycontainer).toBeDefined();
+	// });
 
-	it('should create "imageryCommunicatorService" service is fires event', () => {
+	it('should create "imageryCommunicatorService" service', () => {
```

Comment:
services should tested on separate file (imageryCommunicator.service.ts for example)